src/main/java/com/xceptance/xlt/report/providers/CustomLogsReportProvider.java

```diff
@@ -0,0 +1,217 @@
+ */
+public class CustomLogsReportProvider extends AbstractReportProvider
+{
+
+    private static final String CUSTOM_DATA = "custom_data_logs";
+    
+    private Map<String, ZipOutputStream> foundScopes = new HashMap<String, ZipOutputStream>();
+    
+    private Path baseDir = null;
+    private Path targetDir = null;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Object createReportFragment()
+    {
+        final CustomLogsReport report = new CustomLogsReport();
+
+        FileObject results = ((ReportGeneratorConfiguration) getConfiguration()).getResultsDirectory();
+        
+        baseDir = results.getPath();
+        targetDir = Paths.get(getConfiguration().getReportDirectory() + File.separator + CUSTOM_DATA);       
+        
+        try
+        {
+            findLogs(results);
+        }
+        catch (IOException e)
+        {
+            System.err.println("Failed to walk file tree searching for custom data logs. Cause: " + e.getMessage());
+        }
+        finally 
+        {
+            closeAllStreams(); // this closes the ZipOutputStreams, so this MUST be done
+        }
+
+        // add the link/size info for found scopes
+        Set<String> scopes = getResult();
+        
+        for (String scope : scopes)
+        {    
+            CustomLogReport clr = new CustomLogReport();
+            clr.scope = scope;
+            
+            Path targetPath = Paths.get(targetDir + File.separator + scope + ".zip");
+            
+            try
+            {                
+                clr.size = Files.size(targetPath);
+                clr.path = CUSTOM_DATA + File.separator + scope + ".zip";
+            }
+            catch (IOException e)
+            {
+                System.err.println("Unable to collect information for custom data logs for " + scope + ". Cause: " + e.getMessage());
+            }
+            
+            report.customLogs.add(clr);
+        }
+              
+
+        return report;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void processDataRecord(final Data data)
+    {
+        // nothing to do here
+    }
+
+    /**
+     * Tell the system that there is no need to call processDataRecord
+     */
+    @Override
+    public boolean wantsDataRecords()
+    {
+        return false;
+    }        
+
+    /**
+     * Recursive method to walk the directory tree searching for custom log files. 
+     * Found files are directly put into a zip archive for the corresponding custom data scope in the report.
+     * @param file the directory or file to start from
+     * @throws IOException
+     */
+    private void findLogs (FileObject file) throws IOException
+    {
+        if (file.getType() == FileType.FOLDER && !XltConstants.CONFIG_DIR_NAME.equals(file.getName().getBaseName().toString()))
+        {
+            for (final FileObject fo : file.getChildren())
+            {
+                findLogs(fo);
+            }
+        }
```

Comment:
I would like to avoid combining the file type check with another condition. Please move the second condition to a new `if` inside the body of this `if`.